scripts/imgtool/image.py

```diff
@@ -81,21 +96,32 @@ def load(cls, path, included_header=False, **kwargs):
         obj.check()
         return obj
 
-    def __init__(self, version=None, header_size=IMAGE_HEADER_SIZE, pad=0):
+    def __init__(self, version=None, header_size=IMAGE_HEADER_SIZE, pad=0,
+                 base_addr=None, format=BIN_EXT):
         self.version = version or versmod.decode_version("0")
         self.header_size = header_size or IMAGE_HEADER_SIZE
+        self.base_addr = base_addr
         self.pad = pad
+        self.format = format
```

Comment:
Instead of setting and then checking a string, why not create two save methods, one for bin, and one for hex, and set self format to the desired method. self.save would then just invoke self.format(self, path)?

Reply:
That seems equivalently future-proof but I have no objections to it

Reply:
I can do that yes. I don't have a strong preference for either.

Reply:
Might as well do it properly and use inheritance